tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs

```diff
@@ -280,6 +284,89 @@ internal void UpdateAddedProbeInstrumentations(IReadOnlyList<ProbeDefinition> ad
             }
         }
 
+        private static string[] SafeParseSignature(string signature)
+        {
+            if (string.IsNullOrEmpty(signature))
+            {
+                return null;
+            }
+
+            try
+            {
+                return ParseSignature(signature);
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, "Error parsing the signature {Signature}. Fall back to empty signature", signature);
+                return null;
+            }
+        }
+
+        private static string[] ParseSignature(string signature)
+        {
+            if (signature.StartsWith("(") && signature.EndsWith(")"))
+            {
+                signature = signature.Substring(1, signature.Length - 2);
+            }
+
+            var types = new List<string>();
+            var initialCapacity = 32;
+            var currentType = StringBuilderCache.Acquire(initialCapacity);
+            var stack = new Stack<char>();
+
+            try
+            {
+                foreach (char c in signature)
+                {
+                    if (c == ' ')
+                    {
+                        continue;
+                    }
+
+                    if (c == ',' && stack.Count == 0)
+                    {
+                        types.Add(StringBuilderCache.GetStringAndRelease(currentType));
+                        currentType = StringBuilderCache.Acquire(initialCapacity);
+                    }
+                    else
+                    {
+                        if (c == '<')
```

Comment:
I don't think it would happen in the signatures we deal with 🤔, I mean, accepting an argument that is compiler generated?

Reply:
Yeah, does this need to handle unpronounceable types generated by the compiler? e.g. `<PrivateImplementationDetails>` and `<>z__ReadOnlyArray<T>`? I'd love to see unit tests for these if nothing else, to confirm that we bail-out at worst case, rather than generating something weird.

Similarly, do we need to handle open-generics?

Reply:
The open generic is handled as closed generic.
Refactored around + added unit tests in d09b0d31c3e24c41453e141423608387754c5534.